pkg/core/object/fmt.go

```diff
@@ -161,14 +161,15 @@ func (v *FormatValidator) Validate(obj *object.Object, unprepared bool) error {
 		if !obj.VerifySignature() {
 			return errors.New("could not validate header fields: invalid signature")
 		}
-		if err := obj.VerifyID(); err != nil {
-			return fmt.Errorf("could not validate header fields: invalid identifier: %w", err)
-		}
 
 		if err := validateSignatureKey(obj); err != nil {
 			return fmt.Errorf("could not validate signature key: %w", err)
 		}
 
+		if err := obj.VerifyID(); err != nil {
```

Comment:
That's exactly what should be done first. It's cheap and it's a format violation. If it's broken signatures don't matter.

Reply:
raised